### PR TITLE
libradosstriper: fix format injection vulnerability

### PR DESCRIPTION
**Pull request by chardan**
http://tracker.ceph.com/issues/20240

Signed-off-by: Jesse Williamson <jwilliamson@suse.de>

### CODE REVIEW

**Comment by chardan** [comment]
As suggested, I'm looking for other situations where format strings can leak in from the outside world, but that will take some time.

**Comment by liewegas** [comment]
jenkins test this please

**Comment by chardan** [comment]
I think this patch should address the submitter's situation.

**Comment by dmick** [comment]
ok to test

File: src/libradosstriper/RadosStriperImpl.cc [hunk 1]
```diff
@@ -12,6 +12,8 @@
  *
  */
 
+#include <boost/algorithm/string/replace.hpp>
```
Comment (tchaikov) [inline src/libradosstriper/RadosStriperImpl.cc]: the commit message should look like:
```
Fixes: http://tracker.ceph.com/issues/20240
SIgned-off-by: Stan K <stan's email>
```
and we need to have Stan's email, assuming he agrees to contribute his patch to Ceph.

Comment (smithfarm) [inline src/libradosstriper/RadosStriperImpl.cc]: @chardan And the top line of the commit message could be `libradosstriper: remove format injection vulnerability` . . . the entire commit message might then look like this:

```
libradosstriper: remove format injection vulnerability

Fixes: http://tracker.ceph.com/issues/20240
SIgned-off-by: Stan K <stan's email>
```

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-06-15_17:39:27-rados-wip-kefu-testing---basic-smithi/ all issues are known ones.

**Comment by chardan** [comment]
...waiting to hear back from the author...

**Comment by tchaikov** [comment]
chardan, Stan K already replied over the ticket, see http://tracker.ceph.com/issues/20240

**Comment by chardan** [comment]
Jenkins retest this please.

**Comment by chardan** [comment]
@tchaikov Could you please peek at this? Thanks.

**Pull request merged**